src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -22,6 +23,74 @@ class DatasetConverter(BaseConverter):
         default=None,
     )
 
+    @classmethod
+    @pydantic.validate_call
+    def from_remote_dandiset(
+        cls,
+        dandiset_id: str = pydantic.Field(pattern=r"^\d{6}$"),
+        api_url: str | None = None,
+        token: str | None = None,
+        limit: int | None = None,
+    ) -> typing_extensions.Self | None:
+        """
+        Initialize a converter of a Dandiset to BIDS format.
+
+        Parameters
+        ----------
+        dandiset_id : str
+            The dandiset ID of the Dandiset to be converted.
+        api_url : str, optional
+            The API URL of a custom DANDI instance to use. If not provided, the API URL of the
+            DANDI instance specified by the :envvar:`DANDI_INSTANCE` environment variable
+            is used. If the :envvar:`DANDI_INSTANCE` environment variable is not specified,
+            The API URL of the `"dandi"` DANDI instance is used.
+        token : str, optional
+            The authentication token for accessing the DANDI instance.
+            If not provided, will attempt to read from the environment variable `DANDI_API_KEY` if it exists.
+            This is required for accessing embargoed Dandisets.
+        limit : int, optional
+            If specified, limits the number of sessions to convert.
+            This is mainly useful for testing purposes.
```

Comment:
If this parameter is just for use in Pytest tests and not for a user to do their manual tests. I think we can specify this value in a module level constant, and monkeypatch this constant when running the Pytest tests. In that way, the argument namespace is not polluted.

Reply:
This parameter is not just for use in Pytest

It is useful in general

There are certainly more refined approaches, open to those in future PRs

But not worth holding this up just for that IMO